lib/geokit-rails/acts_as_mappable.rb

```diff
@@ -315,8 +315,11 @@ def substitute_distance_in_where_values(arel, origin, units=default_units, formu
       # Returns the distance SQL using the spherical world formula (Haversine).  The SQL is tuned
       # to the database in use.
       def sphere_distance_sql(origin, units)
-        lat = deg2rad(origin.lat)
-        lng = deg2rad(origin.lng)
+	# "origin" can be a Geokit::LatLng (with :lat and :lng methos), e.g. when using geo_scope
+	# or it can be an ActsAsMappable with customized latitude and longitude methods, e.g. when using distance_sql.
+	lat = deg2rad(origin.respond_to?(:lat) ? origin.lat : origin.send(:"#{lat_column_name}"))
```

Comment:
Line is too long. [90/80]<br>Use 2 (not -5) spaces for indentation.<br>Tab detected.